hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -161,6 +161,13 @@ public class HoodieClusteringConfig extends HoodieConfig {
           + "value will let the clustering job run faster, while it will give additional pressure to the "
           + "execution engines to manage more concurrent running jobs.");
 
+  public static final ConfigProperty<Integer> CLUSTERING_READ_RECORDS_PARALLELISM = ConfigProperty
+      .key("hoodie.clustering.read.records.parallelism")
+      .defaultValue(20)
```

Comment:
Maybe just rename to `hoodie.clustering.parallelism`

Reply:
changed

Reply:
I'm wondering where does the default 20 come from, should we limit the parallelism by default? Do we have similiar option for compaction?

Reply:
We verify that 20 is a relatively stable parameter running job. The param hoodie.clustering.plan.strategy.target.file.max.bytes is 1g which contains many files, we read it with 20 parallelism is enough and no need to increase executor-memory.

We should limit the parallelism by default, this can reduce the number of tasks to a great extent.

Compaction has a similiar use, it parallelize operations with its size.
```
// HoodieCompactor.java
return context.parallelize(operations).map(operation -> compact(
        compactionHandler, metaClient, config, operation, compactionInstantTime, maxInstantTime, instantRange, taskContextSupplier, executionHelper))
        .flatMap(List::iterator);
```

Reply:
cc @boneanxs to take a look~

Reply:
We already have a `hoodie.clustering.max.parallelism` to control how many clustering jobs to submit, now this param looks to control the parallelism when reading per group, and it only works when row writer is disabled. This configuration name still confuses me.

Is there any other way to avoid a new configuration? What abt we directly use `clusteringGroup.getNumOutputFileGroups`

Reply:
Yeah, it looks like the read parallelism for a single `HoodieClusteringGroup`.

Reply:
> We already have a `hoodie.clustering.max.parallelism` to control how many clustering jobs to submit, now this param looks to control the parallelism when reading per group, and it only works when row writer is disabled. This configuration name still confuses me.
> 
> Is there any other way to avoid a new configuration? What abt we directly use `clusteringGroup.getNumOutputFileGroups`

Yes. it limits the read parallelism for single clutering group. `clusteringGroup.getNumOutputFileGroups` is 2 default, it's too small to read a 1g file.

Maybe we can use `hoodie.clustering.rdd.read.parallelism`.

Reply:
cc @boneanxs   This config `oodie.clustering.rdd.read.parallelism` is ok? or whether you have better adivce.

Reply:
What about the sum of `clusteringGroup.getNumOutputFileGroups` for all clustering groups, which is actually the final write parallelism.

Reply:
> What about the sum of `clusteringGroup.getNumOutputFileGroups` for all clustering groups, which is actually the final write parallelism.

Not very friendly. The number of all clustering groups is controlled by `hoodie.clustering.plan.strategy.max.num.groups`. If we increase this config, the read parallelism will be larger. In our case, we generate 100 groups and the read parallelism       will be 200, that is not helpful for reading tasks.

Reply:
changed